feat: add MCP Server Design Guidelines, update mkocs (#158)
awslabs/mcp: DESIGN_GUIDELINES.md

```toml
[project.scripts]
"awslabs.your-mcp-server" = "awslabs.your_mcp_server.server:main"
```

## Package Naming and Versioning

1. **Package Naming**: Follow the established naming pattern:
   - Namespace: `awslabs`
   - Package name: lowercase with hyphens (in pyproject.toml)
   - Python module: lowercase with underscores

   Example:
   ```toml
   # In pyproject.toml
   name = "awslabs.nova-canvas-mcp-server"
   ```

   ```python
   # In Python imports
   from awslabs.nova_canvas_mcp_server import models
   ```

2. **Versioning**: Store version information in `__init__.py`:

   ```python
   # awslabs/your_mcp_server/__init__.py
   """awslabs Your MCP Server."""

   __version__ = "0.1.0"
   ```

3. **Version Synchronization**: Our monorepo `release.py` bumps the patch version upon changes in:
   - `pyproject.toml`
   - `__init__.py` in the package
   - Configure `commitizen` in `pyproject.toml` to update versions automatically

   ```toml
   [tool.commitizen]
   name = "cz_conventional_commits"
   version = "0.0.0"
   tag_format = "$version"
   version_files = [
       "pyproject.toml:version",
       "awslabs/your_mcp_server/__init__.py:__version__"
   ]

>    update_changelog_on_bump = true
> ## License and Copyright Headers

Reviewer: ```suggestion
   update_changelog_on_bump = true
   ```

## License and Copyright Headers
```

# MCP Server Design Guidelines

This document outlines the design guidelines and best practices for developing MCP (Multimodal Code Platform) servers. These guidelines are based on the patterns used in examples like `bedrock-kb-retrieval-mcp-server` and `nova-canvas-mcp-server`.

## Table of Contents

> - [MCP Server Design Guidelines](#mcp-server-design-guidelines)

Reviewer 2: - Should we have a section related to the number of tools/resources/prompts that a server expose ? 
- What about adding high level guidelines on single responsibility, scope and boundaries ?

Reviewer: > * What about adding high level guidelines on single responsibility, scope and boundaries ?

I think this is sufficient for this PR:
https://github.com/awslabs/mcp/pull/158/files#diff-b5b200f91747f396e76372544579adc60344f4c77e067814b30f2edad39f50a1R83

Reviewer: > * Should we have a section related to the number of tools/resources/prompts that a server expose ?

In a followup PR

    parser.add_argument('--sse', action='store_true', help='Use SSE transport')
    parser.add_argument('--port', type=int, default=8888, help='Port to run the server on')

    args = parser.parse_args()

    # Run server with appropriate transport
    if args.sse:
        mcp.settings.port = args.port
        mcp.run(transport='sse')
    else:
        mcp.run()

if __name__ == '__main__':
    main()
```

3. **Package Entry Point**: Configure the entry point in `pyproject.toml`:

```toml
[project.scripts]
"awslabs.your-mcp-server" = "awslabs.your_mcp_server.server:main"
```

## Package Naming and Versioning

1. **Package Naming**: Follow the established naming pattern:
   - Namespace: `awslabs`
   - Package name: lowercase with hyphens (in pyproject.toml)
   - Python module: lowercase with underscores

   Example:
   ```toml
   # In pyproject.toml
   name = "awslabs.nova-canvas-mcp-server"
   ```

   ```python
   # In Python imports
   from awslabs.nova_canvas_mcp_server import models
   ```

2. **Versioning**: Store version information in `__init__.py`:

   ```python
   # awslabs/your_mcp_server/__init__.py
   """awslabs Your MCP Server."""

   __version__ = "0.1.0"
   ```

> 3. **Version Synchronization**: Keep version consistent between:
>    - `pyproject.toml`
>    - `__init__.py` in the package
>    - Configure `commitizen` in `pyproject.toml` to update versions automatically

Reviewer: ```suggestion
3. **Version Synchronization**: Our monrepo bumps the patch version upon changes in:
   - `pyproject.toml`
   - `__init__.py` in the package
   - Configure `commitizen` in `pyproject.toml` to update versions automatically
```

   ```toml
   [tool.commitizen]
   name = "cz_conventional_commits"
   version = "0.0.0"
   tag_format = "$version"
   version_files = [
       "pyproject.toml:version",
       "awslabs/your_mcp_server/__init__.py:__version__"
   ]
   update_changelog_on_bump = true

>    ```
> 

Reviewer: ```suggestion
   ```

   _NOTE: commitizen attempts to use remote tags, and since each package may be a different version, this feature is disabled in our monorepo_
  
```
